javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```
@@ -393,15 +406,18 @@ void emit(CodeWriter codeWriter, String enumName, Set<Modifier> implicitModifier
             }
 
             // Non-static fields.
-            for (FieldSpec fieldSpec : fieldSpecs) {
-                if (fieldSpec.modifiers().contains(Modifier.STATIC)) {
-                    continue;
-                }
-                if (!firstMember) {
-                    codeWriter.emit("\n");
+            // If kind RECORD, ignore generate Non-static field, records remove the need for them.
```

We shouldn't just silently ignore things that we're explicitly configured in the builder. This makes for very confusing behavior because the generated code does not match the provided `TypeSpec`. 

It's fine to just emit these and generate broken code.

---

Isn't it a bit weird to create private fields for a record?

---

There's a lot of ways to generate code that doesn't compile with this library. Instead of playing whack-a-mole trying to catch all the ways to generate invalid code, I'd prefer to push this responsibility onto consumers using a real Java compiler. That way we know for certain that the generated code is correct.

---

+1, the most important thing is that we don't silently ignore inputs. It's better to require inputs to be reasonable in order to produce code that compiles rather than silently change them in ways that may not be expected, even if the result of that change is code that compiles.